code/modules/reagents/chemistry/holder.dm

```diff
@@ -685,138 +731,376 @@
+			if(cached_my_atom)
+				if(!C.required_container)
+					matching_container = TRUE
 				else
-					if(!C.required_container)
+					if(cached_my_atom.type == C.required_container)
 						matching_container = TRUE
-					if(!C.required_other)
-						matching_other = TRUE
-
-				if(required_temp == 0 || (is_cold_recipe && chem_temp <= required_temp) || (!is_cold_recipe && chem_temp >= required_temp))
-					meets_temp_requirement = TRUE
+				if (isliving(cached_my_atom) && !C.mob_react) //Makes it so certain chemical reactions don't occur in mobs
+					matching_container = FALSE
+				if(!C.required_other)
+					matching_other = TRUE
 
-				if(total_matching_reagents == total_required_reagents && total_matching_catalysts == total_required_catalysts && matching_container && matching_other && meets_temp_requirement)
-					possible_reactions  += C
+				else if(istype(cached_my_atom, /obj/item/slime_extract))
+					var/obj/item/slime_extract/M = cached_my_atom
 
-		if(possible_reactions.len)
-			var/datum/chemical_reaction/selected_reaction = possible_reactions[1]
-			//select the reaction with the most extreme temperature requirements
-			for(var/V in possible_reactions)
-				var/datum/chemical_reaction/competitor = V
-				if(selected_reaction.is_cold_recipe) //if there are no recipe conflicts, everything in possible_reactions will have this same value for is_cold_reaction. warranty void if assumption not met.
-					if(competitor.required_temp <= selected_reaction.required_temp)
-						selected_reaction = competitor
+					if(M.Uses > 0) // added a limit to slime cores -- Muskets requested this
+						matching_other = TRUE
+			else
+				if(!C.required_container)
+					matching_container = TRUE
+				if(!C.required_other)
+					matching_other = TRUE
+
+			if(required_temp == 0 || (is_cold_recipe && chem_temp <= required_temp) || (!is_cold_recipe && chem_temp >= required_temp))
+				meets_temp_requirement = TRUE
+
+			if(total_matching_reagents == total_required_reagents && total_matching_catalysts == total_required_catalysts && matching_container && matching_other && meets_temp_requirement)
+				possible_reactions  += C
+
+	update_previous_reagent_list()
+	//This is the point where we have all the possible reactions from a reagent/catalyst point of view, so we set up the reaction list
+	for(var/pr in possible_reactions)
+		var/datum/chemical_reaction/selected_reaction = pr
+		debug_world("Adding [selected_reaction.type]")
+		if((selected_reaction.reaction_flags & REACTION_INSTANT) || (flags & INSTANT_REACT)) //If we have instant reactions, we process them here
+			instant_react(selected_reaction)
+			.++
+			update_total()
+			continue
+		else
+			var/exists = FALSE
+			for(var/e in reaction_list)
+				var/datum/equilibrium/E_exist = e
+				debug_world("This exists already: [E_exist.reaction.type]")
+				if(E_exist.reaction.type == selected_reaction.type) //Don't add duplicates
+					debug_world("Selected reactions are the same.")
+					exists = TRUE
```

Comment:
```suggestion
				var/datum/equilibrium/E_exist = e
				#ifdef TESTING
				debug_world("This exists already: [E_exist.reaction.type]")
				#endif
				if(E_exist.reaction.type == selected_reaction.type) //Don't add duplicates
					#ifdef TESTING
					debug_world("Selected reactions are the same.")
					#endif
					exists = TRUE
```

Reply:
I completely forgot about the debug_worlds I added

Reply:
I removed them instead